src/Entities/Pawn.cpp

```diff
@@ -111,6 +111,37 @@ void cPawn::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 void cPawn::KilledBy(TakeDamageInfo & a_TDI)
 {
 	ClearEntityEffects();
+
+	if (IsPlayer())
+	{
+		auto Player = static_cast<cPlayer *>(this);
+		cInventory & inv = Player->GetInventory();
+
+		//  If the player is holding a totem of undying in their off-hand or
+		//  main-hand slot and receives otherwise fatal damage, the totem saves the player from death.
+		if ((inv.GetShieldSlot().m_ItemType == E_ITEM_TOTEM_OF_UNDYING ||
+			 inv.GetEquippedItem().m_ItemType == E_ITEM_TOTEM_OF_UNDYING) &&
+				a_TDI.DamageType != dtAdmin && a_TDI.DamageType != dtInVoid
+			)
+		{
+			inv.SetShieldSlot(cItem());
+			m_World->BroadcastEntityStatus(*this, esTotemOfUndying);
+
+			AddEntityEffect(cEntityEffect::effAbsorption, 100, 1);
+			AddEntityEffect(cEntityEffect::effRegeneration, 900, 1);
+			AddEntityEffect(cEntityEffect::effFireResistance, 800, 0);
```

Comment:
Should all effects be cleared as well?

Reply:
They are already cleared on first line of cPawn::KilledBy.